### PR TITLE
feat: typescript 2.9 support

### PR DESCRIPTION
**Pull request by IgorMinar**
known issues:

- [ ] one of the e2e tests are failing because of somehow we don't generate proper amd modules within bazel pipeline. This might be a tsickle 0.30.0 issues. I'm not sure. @alexeagle could you help plz?
- [x] one of the tests for the compiler's metadata compiler is failing. it seems to me that the new behavior is actually correct, but I'd like a clarification from @alxhub plz
- [x] there is one more failing test from the runtime ~I haven't investigated this yet.~  - the issue identified with a runtime error due to undefined  "selector" property seems to be a breakage in the master/HEAD and unrelated to my change. 
- [x] the pr needs to be reformatted

### CODE REVIEW

**Comment by IgorMinar** [comment]
@fyi: @mprobst

File: packages/compiler-cli/test/metadata/collector_spec.ts [hunk 1]
```diff
@@ -974,7 +974,7 @@ describe('Collector', () => {
     });
   });
 
-  describe('regerssion', () => {
+  describe('regeression', () => {
```
Comment (JoostK) [inline packages/compiler-cli/test/metadata/collector_spec.ts]: Haha, nice one but still incorrect :)

**Comment by EatonZ** [comment]
Just an FYI.. you have 2.10 in a few spots here. The next version is actually [3.0.](https://github.com/Microsoft/TypeScript/wiki/Roadmap)

**Comment by mprobst** [comment]
@IgorMinar what's the status of this? We'd like to move google3 to TS2.9 and are blocked on this.

**Comment by mhevery** [comment]
Since @IgorMinar is on vacation, I will take over this PR.

File: packages/compiler-cli/test/metadata/collector_spec.ts (hunk 1 above)
Comment (mhevery) [reply packages/compiler-cli/test/metadata/collector_spec.ts]: fixed

**Review by mhevery — APPROVED** [review]

**Comment by mhevery** [comment]
Blocked on https://github.com/angular/tsickle/issues/846

**Comment by mhevery** [comment]
http://test/OCL:202553051:BASE:202553124:1530224264650:edb25536

File: packages/common/locales/BUILD.bazel [hunk 2]
```diff
@@ -22,6 +22,10 @@ npm_package(
         # We don't actually import anything in the locale code so we can
         # null out the require reference passed into the module.
         "factory\(require, exports\)": "factory(null, exports)",
+
+        # Workaround for `.d.ts`` containing `/// <amd-module .../>`
```
Comment (alxhub) [inline packages/common/locales/BUILD.bazel]: This is not strictly the case - they were generated in ts < 2.9, but not validated?

Comment (alxhub) [reply packages/common/locales/BUILD.bazel]: Ah, it was tsickle adding them before.

**Review by alxhub — APPROVED** [review]

File: packages/compiler/test/aot/compiler_spec.ts [hunk 3]
```diff
@@ -326,7 +326,7 @@ describe('compiler (unbundled Angular)', () => {
     expect(genSource.startsWith(genFilePreamble)).toBe(true);
   });
 
-  it('should be able to use animation macro methods', () => {
+  fit('should be able to use animation macro methods', () => {
```
Comment (eduardocereto) [inline packages/compiler/test/aot/compiler_spec.ts]: Was this a typo?

**Review by eduardocereto — CHANGES_REQUESTED** [review]

**Comment by EatonZ** [comment]
Was I wrong about replacing 2.10 with 3.0?

File: packages/compiler/test/aot/compiler_spec.ts (hunk 3 above)
Comment (mhevery) [reply packages/compiler/test/aot/compiler_spec.ts]: opps! thank for noticing

**Comment by mhevery** [comment]
> Was I wrong about replacing 2.10 with 3.0?

Could you give me more context?

**Comment by mhevery** [comment]
> Was I wrong about replacing 2.10 with 3.0?

I see you are saying that it `package.json` the typescript should be `3.0`. Actually Typescript does not follow semver so they can do any version they want. Having said that `"typescript": ">=2.7.2 <2.10"` is specifically saying that we are only supporting `2.7.2 thorough 2.8.x`. Wether they release 3.0 or 2.9 we will have to do active work to support it, hence the restrictive version.

Do you have a different point of view?

**Comment by EatonZ** [comment]
It was just my thought since there will not be a 2.10, it might be more correct to make 3.0 the max version for your checks. I believe the current restrictions will work fine as-is, but it may be more readable to people looking at it in the future if 3.0 was used over 2.10.

**Comment by mgol** [comment]
I agree with Misko that limiting by 2.10 is the correct choice here. Angular shouldn’t assume anything more than necessary about future TypeScript releases. The TypeScript team may not plan to release 2.10 but they can always change their decision.

Angular only knows it works with TypeScript 2.9.x or older (with some restrictions). The way to express this contract in semver is "<2.10" or "<=2.9", not "<3.0".

**Comment by mhevery** [comment]
http://test/OCL:203143205:BASE:203143231:1530638557253:a92f7ca2

File: integration/typings_test_ts29/package.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    "@angular/platform-browser-dynamic": "file:../../dist/packages-dist/platform-browser-dynamic",
+    "@angular/platform-server": "file:../../dist/packages-dist/platform-server",
+    "@angular/platform-webworker": "file:../../dist/packages-dist/platform-webworker",
+    "@angular/platform-webworker-dynamic": "file:../../dist/packages-dist/platform-webworker-dynamic",
+    "@angular/router": "file:../../dist/packages-dist/router",
+    "@angular/service-worker": "file:../../dist/packages-dist/service-worker",
+    "@angular/upgrade": "file:../../dist/packages-dist/upgrade",
+    "@types/jasmine": "2.5.41",
+    "rxjs": "file:../../node_modules/rxjs",
+    "typescript": "2.8.x",
```
Comment (cexbrayat) [inline integration/typings_test_ts29/package.json]: @mhevery Is it not weird to have `2.8.x` in the package.json of `typings_test_ts29`. Shouldn't that be `2.9.x` (and in the description above)

Comment (mhevery) [reply integration/typings_test_ts29/package.json]: good catch, and thanks for the fix.

**Pull request closed without merging**